### PR TITLE
fix(compiler): auto declare `entryComponents` recursively

### PR DESCRIPTION
**Pull request by tbosch**

### CODE REVIEW

File: modules/@angular/compiler/src/metadata_resolver.ts [hunk 1]
```diff
@@ -370,6 +352,11 @@ export class CompileMetadataResolver {
             `NgModule ${stringify(moduleMeta.type.runtime)} uses ${stringify(entryComponentType.runtime)} via "entryComponents" but it was neither declared nor imported! This warning will become an error after final.`);
       }
     });
+    // Collect @Component.directives/pipes/entryComponents into our declared
+    // directives/pipes. Do this last so that directives added by previous steps
+    // arre considered as well!
```
Comment (vicb) [inline modules/@angular/compiler/src/metadata_resolver.ts]: "are" typo

File: modules/@angular/compiler/src/metadata_resolver.ts [hunk 2]
```diff
@@ -396,7 +380,8 @@ export class CompileMetadataResolver {
             `Unexpected pipe value '${pipeType}' on the View of component '${stringify(compMeta.type.runtime)}'`);
       }
       const pipeMeta = this.getPipeMetadata(pipeType);
-      this._addPipeToModule(pipeMeta, moduleType, transitiveModule, declaredPipes);
+      this._addPipeToModule(
```
Comment (vicb) [inline modules/@angular/compiler/src/metadata_resolver.ts]: given the name `_addPipeToModule` what about `this._addPipeToModule(pipe, module)` ? same below

Comment (tbosch) [inline modules/@angular/compiler/src/metadata_resolver.ts]: Won't work, as I don't have the `CompileNgModuleMetadata` yet, only the `CompileTransitiveNgModuleMetadata`...

Comment (vicb) [inline modules/@angular/compiler/src/metadata_resolver.ts]: `moduleMeta.type.runtime, moduleMeta.transitiveModule, moduleMeta.declaredPipes` -> `moduleMeta` ??

Comment (tbosch) [inline modules/@angular/compiler/src/metadata_resolver.ts]: Sorry, I meant in this place I actually have it, but at another place I don't (as I am in the process of creating it).

**Pull request closed without merging**